packages/kbn-apm-config-loader/src/config.test.ts

```diff
@@ -361,5 +361,49 @@ describe('ApmConfiguration', () => {
         })
       );
     });
+
+    it('allows overriding some services settings', () => {
+      const kibanaConfig = {
+        elastic: {
+          apm: {
+            active: true,
+            serverUrl: 'http://an.internal.apm.server:port/',
+            transactionSampleRate: 0.1,
+            servicesOverrides: [
+              {
+                name: 'externalServiceName',
+                active: false,
+                serverUrl: 'http://a.public.apm.server:port/',
+                disableSend: true, // just adding an extra field to prove merging works
+              },
+            ],
+          },
+        },
+      };
+
+      const internalService = new ApmConfiguration(mockedRootDir, kibanaConfig, true).getConfig(
+        'internalServiceName'
+      );
+      expect(internalService).toEqual(
+        expect.objectContaining({
+          active: true,
+          serverUrl: 'http://an.internal.apm.server:port/',
+          transactionSampleRate: 0.1,
+        })
+      );
+      expect(internalService).not.toHaveProperty('disableSend');
+      expect(internalService).not.toHaveProperty('servicesOverrides'); // We don't want to leak this to the client's config
+
+      expect(
+        new ApmConfiguration(mockedRootDir, kibanaConfig, true).getConfig('externalServiceName')
+      ).toEqual(
+        expect.objectContaining({
+          active: false,
+          serverUrl: 'http://a.public.apm.server:port/',
+          transactionSampleRate: 0.1,
+          disableSend: true,
```

Comment:
I would check for the value of `serviceName` just to be sure:

```suggestion
          disableSend: true,
          serviceName: 'externalServiceName',
```